pyout/field.py

```diff
@@ -42,7 +46,43 @@ def __init__(self, width=10, align="left"):
         self._align = align
         self._fmt = self._build_format()
 
-        self.processors = {"core": [], "default": []}
+        self.pre = {}
+        self.pre_keys = []
+        self.post = {}
+        self.post_keys = []
+
+    def add(self, kind, key, *values):
+        """Add processor functions.
+
+        This method both adds the processor function and registers the
+        key.  As a result, any processor added through this method
+        will be enabled when the instance is called without the
+        `pre_keys` (if kind is "pre") or `post_keys` argument (if kind
+        is "post").  To set up non-default keys, modify the `pre` or
+        `post` attributes directly.
+
+        Parameters
+        ----------
+        kind : {"pre", "post"}
+        key : str
+            Put the functions under this key in the pre or post
+            processor collection.
```

Comment:
what is the usecase for those "keys", i.e. why those (pre/post) are just lists?
since they are dicts, wouldn't it make order of application nondeterministic?

Reply:
`__call__` accesses the keys in the order of `pre_keys`/`post_keys`,
so it's deterministic.

A dict of lists rather than a list of dicts is used so that the
`pre_keys`/`post_keys` arguments can be used to select certain sets.
I touch on this a bit in the [class docstring][0], but I can add more
about that design decision there.

In the case of Tabular, it usually uses "core"/"default" but for
overriding styles it uses "core"/"override".

https://github.com/pyout/pyout/blob/54c6ec9e25ce652f44035222b99cde8c7955d5b4/pyout/tabular.py#L197-L207

[0]: https://github.com/pyout/pyout/blob/54c6ec9e25ce652f44035222b99cde8c7955d5b4/pyout/field.py#L35

Reply:
so those 'keys' are from predefined lists (e.g. `['core', 'default', 'override']`).
Then shouldn't in https://github.com/pyout/pyout/blob/54c6ec9e25ce652f44035222b99cde8c7955d5b4/pyout/field.py#L80
instead of extending a list of keys, verify that they key is among explicitly 'registered'?